extensions/nns/src/install_nns.rs

```diff
@@ -416,38 +413,24 @@ pub struct IcNnsInitOpts {
 ///   - This won't work with an HSM, because the agent holds a session open
 ///   - The provider_url is what the agent connects to, and forwards to the replica.
 #[context("Failed to install NNS components.")]
-pub async fn ic_nns_init(ic_nns_init_path: &Path, opts: &IcNnsInitOpts) -> anyhow::Result<()> {
-    let mut cmd = std::process::Command::new(ic_nns_init_path);
-    cmd.arg("--pass-specified-id");
-    cmd.arg("--url");
-    cmd.arg(&opts.nns_url);
-    cmd.arg("--wasm-dir");
-    cmd.arg(&opts.wasm_dir);
-    opts.test_accounts.iter().for_each(|account| {
-        cmd.arg("--initialize-ledger-with-test-accounts");
-        cmd.arg(account);
-    });
-    opts.sns_subnets.iter().for_each(|subnet| {
-        cmd.arg("--sns-subnet");
-        cmd.arg(subnet);
-    });
-    let args: Vec<_> = cmd
-        .get_args()
-        .into_iter()
-        .map(OsStr::to_string_lossy)
-        .collect();
-    dbg!(&cmd);
-    println!("ic-nns-init {}", args.join(" "));
-    cmd.stdout(std::process::Stdio::inherit());
-    cmd.stderr(std::process::Stdio::inherit());
-    let output = cmd
-        .output()
-        .with_context(|| format!("Error executing {:#?}", cmd))?;
-
-    if !output.status.success() {
-        return Err(anyhow!("ic-nns-init call failed"));
+pub async fn ic_nns_init(opts: &IcNnsInitOpts, dfx_cache_path: &Path) -> anyhow::Result<String> {
+    let mut args: Vec<OsString> = vec![
+        "--pass-specified-id".into(),
+        "--url".into(),
+        opts.nns_url.clone().into(),
+        "--wasm-dir".into(),
+        opts.wasm_dir.as_os_str().into(),
+    ];
+    for account in &opts.test_accounts {
+        args.push("--initialize-ledger-with-test-accounts".into());
+        args.push(account.into());
     }
-    Ok(())
+    if let Some(subnets) = &opts.sns_subnets {
+        args.push("--sns-subnet".into());
+        args.push(subnets.into());
+    }
+    call_extension_bundled_binary(dfx_cache_path, "ic-nns-init", &args)
+        .with_context(|| format!("Error executing `ic-nns-init` with args: {:?}", args))
```

Comment:
The error output for `call_extension_bundled_binary` already includes the arguments, yes?